src/state.ts

```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import { DetailedMovie } from "./API";
+import produce from "immer";
+
+// the "as const" is just saying this is an array of these specific strings
+export const mediaTypes = ["movie", "series"] as const;
+// because mediaTypes is an array of 2 (currently but could be updated) specific strings
+// if we get it's type when accessed by a number
+// e.g. mediaTypes[0] we get a string union e.g. "movie" | "series" without having to manually type it
+export type MediaType = typeof mediaTypes[number];
+
+export type State = {
+  search: string;
+  movies: DetailedMovie[];
+  page: number;
+  selectedMovie: string | undefined;
+  popupState: boolean;
+  mediaType: MediaType;
+};
+
```

Comment:
I like the choice of adding a file completely dedicated to state, how do you decide whether or not to store state in here?

Reply:
i'd put all app state in the reducer, if you had a small bit of state that you think won't be affected by any other state then you could still use `useState` but i'd recommend only if that is in a component below App.tsx